src/EcoSurveyComponent.java

```diff
@@ -0,0 +1,147 @@
+import components.map.Map;
+import components.map.Map1L;
+import components.simplereader.SimpleReader;
+import components.simplereader.SimpleReader1L;
+import components.simplewriter.SimpleWriter;
+import components.simplewriter.SimpleWriter1L;
+
+public class EcoSurveyComponent {
+    private final class Species {
```

Comment:
Love it! An alternative to this might be a Map. Saves you from having to update the internal class with extra fields. Though, it does require you to use strings or enums as keys.